docs/fundamentals/environments.rst

```diff
@@ -1,8 +1,120 @@
+
+The following ``StartupDevelopment`` file from this articles sample project is run when the application is set to run in a Development environment:
+
+.. literalinclude:: environments/sample/src/Environments/StartupDevelopment.cs
+	:language: c#
+	:caption: StartupDevelopment.cs
+	:linenos:
+	:emphasize-lines: 5
+
+Run the application in development, and a welcome screen is displayed. The sample also includes a ``StartupStaging`` class:
+
+.. literalinclude:: environments/sample/src/Environments/StartupStaging.cs
+	:language: c#
+	:caption: StartupStaging.cs
+	:linenos:
+	:emphasize-lines: 6
+
+When the application is run with ``ASPNET_ENV`` set to ``Staging``, this ``Startup`` class is used, and the application will simply display a string stating it's running in a staging environment. The application's default ``Startup`` class will only run when the environment is not set to either ``Development`` or ``Staging`` (presumably, this would be when it is set to ``Production``, but you're not limited to only these three options. Also note that if no environment is set, the default ``Startup`` will run).
+
+In addition to using an entirely separate ``Startup`` class based on the current environment, you can also make adjustments to how the application is configured within a ``Startup`` class. The ``Configure()`` and ``ConfigureServices()`` methods support environment-specific versions similar to the ``Startup`` class itself, of the form ``Configure[Environment]()`` and ``Configure[Environment]Services()``. If you define a method ``ConfigureDevelopment()`` it will be called instead of ``Configure()`` when the environment is set to development. Likewise, ``ConfigureDevelopmentServices()`` would be called instead of ``ConfigureServices()`` in the same environment.
+
+Determining the environment at runtime
+--------------------------------------
+
+The ``IHostingEnvironment`` service provides the core abstraction for working with environments. This service is provided by the ASP.NET hosting layer, and is injected if you request an instance of ``IHostingEnvironment`` in the parameter list of the ``Configure()`` and ``ConfigureServices()`` methods within ``Startup``. The ASP.NET 5 web site template in Visual Studio uses this approach to load environment-specific configuration files (if present) and to customize the app's error handling settings. In both cases, this behavior is achieved by referring to the currently specified environment by calling ``EnvironmentName`` or ``IsEnvironment`` on the instance of ``IHostingEnvironment`` passed into the appropriate method.
```

Comment:
It can also be injected into your Startup class constructor and other places in your application. Just say it is available via DI.